.github/workflows/build-and-runtime-test.yml

```
@@ -133,3 +133,19 @@ jobs:
           AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID_METRIC_LOGGER }}
           AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY_METRIC_LOGGER }}
           AWS_REGION: us-east-2
+
+  upload-files-on-error:
+    #send video and screenshots on failure as an artifact to be checked later
+    runs-on: ubuntu-latest
+    environment: ci
+    needs: build
+    if: ${{ failure() }}
+    steps:
+      - name: Upload videos and screenshots
+        uses: actions/upload-artifact@v3
+        with:
+          name: build-and-runtime-test-cypress-error-files
+          path: |
+            canary/e2e/cypress/videos/sign*/**
+            canary/e2e/cypress/screenshots/sign*/**
```

NIT:
```suggestion
            canary/e2e/cypress/videos/**
            canary/e2e/cypress/screenshots/**
```

so that it's independent from test names?

---

ok!

---

dccc3fc